bark/geometry/line.hpp

```diff
@@ -255,6 +255,10 @@ inline bool CheckSForSegmentIntersection(Line l, float s) {
   return start_it != start_it_low;
 }
 
+inline Point2d GetPointAtIdx(const Line& l, int idx) {
+  return l.obj_.at(idx);
```

Comment:
this is an unsafe access!
check if idx exists, otherwise it might crash.

Reply:
If idx > l._obj.size() maybe return the last point? otherwise the first? add log message for that as well